File: python/sorting/bubble_sort.py
data = [ 5, 10, 11, 3, 4, 18, 7, 21 ];

def bubble_sort(arr):
    for i in range(len(arr)):
        for j in range (0, (len(arr))-i-1):
            if arr[j] > arr[j+1]:
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
    return arr

print(bubble_sort(data))

File: java/sorting/BubbleSort.java
package sorting;

import java.util.Arrays;

public class BubbleSort {
    public static void main(final String[] args) {
        int[] array = new int[] { 5, 6, 7, 2, 4, 1, 7 };
        sort(array);

        System.out.println(Arrays.toString(array));
    }

    public static void sort(int[] arr) {
        Boolean isSorted;
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            isSorted = true;
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    // swap arr[j+1] and arr[i]
                    swap(arr, j, j + 1);
                    isSorted = false;
                }
            }

            // if a pass completes without swapping, break loop
            if (isSorted) {
                break;
            }
        }
    }

    public static void swap(int[] arr, int firstIdx, int secondIdx) {
        int temp = arr[firstIdx];
        arr[firstIdx] = arr[secondIdx];
        arr[secondIdx] = temp;
    }
}